content/opt/setup/50-icinga2

```diff
@@ -28,15 +28,34 @@ fi
 
 chfn -f "${ICINGA2_USER_FULLNAME}" nagios
 
-# MySQL installation
-mysql <<-END
-  CREATE DATABASE IF NOT EXISTS icinga;
-  GRANT SELECT, INSERT, UPDATE, DELETE, DROP, CREATE VIEW, INDEX, EXECUTE ON icinga.* TO 'icinga'@'localhost' IDENTIFIED BY '${ICINGA_PASSWORD}';
-END
+# Create MySQL user on localhost or test given connection
+if [ "${ICINGA2_IDO_MYSQL_HOST}" == "localhost" ]; then
+	mysql <<-END
+	  CREATE DATABASE IF NOT EXISTS ${ICINGA2_IDO_MYSQL_DATA};
+	  GRANT ALL
+	    ON ${ICINGA2_IDO_MYSQL_DATA}.*
+	    TO '${ICINGA2_IDO_MYSQL_USER}'@'${ICINGA2_IDO_MYSQL_HOST}'
+	    IDENTIFIED BY '${ICINGA2_IDO_MYSQL_PASS}';
+	END
+else
+	mysql \
```

Comment:
This check for DB access is quite senseless, if we can't `exit 1` and `run-parts` will also exit with 1 if one of the scripts fail.
Also we might want to check for

```
SHOW GRANTS;
SHOW GRANTS FOR CURRENT_USER;
SHOW GRANTS FOR CURRENT_USER();
```

That we actually can seed the DB.